modules/zip/src/lib/tar/tar.ts

```diff
@@ -7,42 +7,39 @@
 
 import {clean, pad, stringToUint8} from './utils';
 import {format} from './header';
+import type {Blocks, Options, Chunks, Chunk} from './types';
 
-let blockSize;
+let blockSize: number;
 const recordSize = 512;
 
 class Tar {
-  /**
-   * @param {number} [recordsPerBlock]
-   */
-  constructor(recordsPerBlock) {
+  written: number;
+  out: Uint8Array;
+  blocks: Blocks;
+  length: number;
+
+  constructor(recordsPerBlock: number) {
     this.written = 0;
     blockSize = (recordsPerBlock || 20) * recordSize;
     this.out = clean(blockSize);
-    /** @type {{ header: any; input: string | Uint8Array; headerLength: number; inputLength: number; }[]} */
     this.blocks = [];
     this.length = 0;
     this.save = this.save.bind(this);
     this.clear = this.clear.bind(this);
     this.append = this.append.bind(this);
   }
 
-  /**
```

Comment:
We should not remove JSDoc headers. Just remove the type info inside `{}` brackets. We can still list the params and provide explanations for them.